### PR TITLE
Restructure all docs into JupyterBook format

### PR DESCRIPTION
**Pull request by hyanwong**
And fix some internal references

### CODE REVIEW

**Comment by hyanwong** [comment]
For docs to appear on the tskit web site, need to add tsdate into https://github.com/tskit-dev/tskit-site/blob/main/.github/workflows/build-deploy.yml

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+queue_rules:
+  - name: default
+    conditions:
+      - "#approved-reviews-by>=1"
+      - "#changes-requested-reviews-by=0"
+      - status-success=Lint
+      - status-success=Python (3.8, macos-latest)
```
Comment (benjeffery) [inline .mergify.yml:7]: `tsdate` doesn't have github actions tests, so these merge conditions will never be met. 
You just need:

      - "status-success=ci/circleci: build"
      - status-success=Docs

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Build Docs
+        run: |
+          make -C docs
+
+      - name: Trigger docs site rebuild
+        if: github.ref == 'refs/heads/main'
+        run: |
+          curl -X POST https://api.github.com/repos/tskit-dev/tskit-site/dispatches \
+                    -H 'Accept: application/vnd.github.everest-preview+json' \
+                    -u AdminBot-tskit:${{ secrets.ADMINBOT_TOKEN }} \
```
Comment (benjeffery) [inline .github/workflows/docs.yml:47]: tsdate repo will need the adminbot token added to it's secrets for this - I'll action

File: .mergify.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+pull_request_rules:
+  - name: Automatic rebase, CI and merge
+    conditions:
+      - "-merged"
+      - "#approved-reviews-by>=1"
+      - "#changes-requested-reviews-by=0"
+      - base=main
+      - label=AUTOMERGE-REQUESTED
+      - status-success=Lint
+      - status-success=Python (3.8, macos-latest)
```
Comment (benjeffery) [inline .mergify.yml:24]: Same here

**Review by benjeffery — APPROVED** [review]
Some issues with the mergify config, otherwise great.
I haven't been line by line through all the docs files - I assume there are no changes there that aren't just for JB.
Thanks @hyanwong I'll get on with the site action.

File: .mergify.yml (hunk 3 above)
Comment (hyanwong) [reply .mergify.yml:24]: Ah, sorry - this answers my question above!

**Comment by hyanwong** [comment]
Thanks for looking @benjeffery - I have changed the mergify.yl file - I'm not 100% sure what I'm doing there, so I hope I have it right. I'm pretty sure that the unit tests are run when PRs are made, but I'm assuming mergify is looking at something else.

File: .mergify.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+queue_rules:
+  - name: default
+    conditions:
+      - "#approved-reviews-by>=1"
+      - "#changes-requested-reviews-by=0"
+      - status-success=Lint
```
Comment (benjeffery) [inline .mergify.yml:6]: There's no lint action so that line needs to be removed here and below.

**Review by benjeffery — COMMENTED** [review]
Almost there - the tsdate repo just runs tests on circleCI not GitHub actions hence the removal of all these lines.

**Review by benjeffery — APPROVED** [review]

**Pull request merged**